client/store.js

```diff
@@ -55,6 +55,16 @@ store.getCurrentAccountId = function() {
     return data.currentAccountId;
 }
 
+store.getDefaultAccountId = function() {
+    return data.settings.get('defaultAccountId');
+}
+
+store.setDefaultAccountId = function(id) {
+    //Do not use store.changeSetting so that the change is immediately visible
```

Comment:
I like the optimistic version better : as soon as I click the star becomes blue, whereas with this Promise, the icon will only become blue after everything was set on the server.

Since setting an account is a small task from a user pov, I find it confusing to have to wait for the server response.
What about I do something like this : 

```js
store.changeSetting = function(key, value) {
    var previousValue = data.settings.get(key);
    data.settings.set(key, value);

    backend.saveSetting(String(key), String(value))
    .catch(err => {
        GenericErrorHandler(err);
        data.settings.set(key, previousValue);
        flux.dispatch({
            type: Events.forward,
            event: State.defaultAccount
        });
    });
}
```